AzureMonitor: Update documentation and v8 Whats New (#35275)
grafana/grafana: docs/sources/datasources/azuremonitor.md

The Azure Monitor data source supports multiple services in the Azure cloud:

- **[Azure Monitor Metrics]({{< relref "#query-the-metrics-service" >}})** (or Metrics) is the platform service that provides a single source for monitoring Azure resources.

> - **[Azure Monitor Logs]({{< relref "#query-the-logs-service" >}})** (or Logs) gives you access to log data collected by Azure Monitor. -**[Azure Monitor Logs]({{< relref "#query-the-azure-resource-graph-service" >}})** (or Logs) gives you access to log data collected by Azure Monitor.

Reviewer: It looks like you have this Azure Monitor Logs part twice with 2 different links is that intentional?

Author: Good catch!

> ## Add the data source

Reviewer 2: ```suggestion
## Access Azure metrics
```

Reviewer 2: I do not see this change, I had requested earlier.

Author: I'm not too sure this change makes sense? This section is about adding and configuring Azure Monitor

> The data source can access metrics from four different services. You can configure access to the services that you use. It is also possible to use the same credentials for multiple services if that is how you have set it up in Azure AD.

Reviewer 2: ```suggestion
The Azure Monitor data source can access metrics from four different services. Configure access to the services that you plan to use. You can use the same credentials for multiple services if that is how you have set it up in Azure AD.
```

Author: Ahh, this is actually slightly out of date so I'll change this to 

> The Azure Monitor data source can access metrics from three different services. Configure access to the services that you plan to use. To use different credentials for different Azure services, configure multiple Azure Monitor data sources.

> 

Reviewer 2: Here are some links on how to set up access to the different services:

- [Guide to setting up an Azure Active Directory Application for Azure Monitor.](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal)
- [Guide to setting up an Azure Active Directory Application for Azure Monitor Logs.](https://dev.loganalytics.io/documentation/Authorization/AAD-Setup)

> 1. Accessed from the Grafana main menu, newly installed data sources can be added immediately within the Data Sources section. Next, click the "Add data source" button in the upper right. The Azure Monitor data source will be available for selection in the Cloud section in the list of data sources.

Reviewer 2: Rewite this following Prometheus data source doc.

Author: I'm holding off on the bigger refresh, but rather just making minimum updates to make sure it's accurate for the changes in 8.0.

Next week we'll take a look at refreshing the entire documentation.

Reviewer 2: Ok.

2. In the name field, Grafana will automatically fill in a name for the data source - `Azure Monitor` or something like `Azure Monitor - 3`. If you are going to configure multiple data sources, then change the name to something more informative.

3. If you are using Azure Monitor, you need 4 pieces of information from the Azure portal (see link above for detailed instructions):

   - **Tenant Id** (Azure Active Directory -> Properties -> Directory ID)
   - **Client Id** (Azure Active Directory -> App Registrations -> Choose your app -> Application ID)
   - **Client Secret** (Azure Active Directory -> App Registrations -> Choose your app -> Keys)

>    - **Default Subscription Id** (Subscriptions -> Choose subscription -> Overview -> Subscription ID)

Reviewer 2: All this information should ideally be in a table.

4. Paste these four items into the fields in the Azure Monitor API Details section:
   {{< figure src="/static/img/docs/v62/config_1_azure_monitor_details.png" class="docs-image--no-shadow" caption="Azure Monitor Configuration Details" >}}

   - The Subscription Id can be changed per query. Save the data source and refresh the page to see the list of subscriptions available for the specified Client Id.

> 5. If you are also using the Azure Monitor Logs service, then you need to specify these two configuration values (or you can reuse the Client Id and Secret from the previous step).

Author: Good catch. Removed this 😎

   - Client Id (Azure Active Directory -> App Registrations -> Choose your app -> Application ID)
   - Client Secret (Azure Active Directory -> App Registrations -> Choose your app -> Keys -> Create a key -> Use client secret)

> 6. Test that the configuration details are correct by clicking on the "Save & Test" button:

Reviewer 2: ```suggestion
1. Test that the configuration details are correct by clicking on the "Save & Test" button:
```

Author: I think `6.` is correct - this is the 6th step in the series, no?

Reviewer 2: All numbered steps should be "1.", they get converted.

## Choose a Service

In the query editor for a panel, after choosing your Azure Monitor data source, the first option is to choose a service. There are three options here:

- Metrics
- Logs
- Azure Resource Graph

The query editor changes depending on which one you pick. Metrics is the default.

> Starting in Grafana 7.4, the Azure Monitor query type was renamed to Metrics and Azure Logs Analytics was renamed to Logs. Starting in Grafana 8.0 Application Insights and Insights Analytics is unavailable for new panels, in favor of querying through Metrics and Logs.

Reviewer 2: ```suggestion
In Grafana 7.4, the Azure Monitor query type was renamed to Metrics, and Azure Logs Analytics was renamed to Logs. In Grafana 8.0 Application Insights and Insights Analytics is unavailable for new panels, in favor of querying through Metrics and Logs.
```

## Choose a Service

In the query editor for a panel, after choosing your Azure Monitor data source, the first option is to choose a service. There are three options here:

- Metrics
- Logs
- Azure Resource Graph

The query editor changes depending on which one you pick. Metrics is the default.

> Starting in Grafana 7.4, the Azure Monitor query type was renamed to Metrics and Azure Logs Analytics was renamed to Logs. Starting in Grafana 8.0 Application Insights and Insights Analytics is unavailable for new panels, in favor of querying through Metrics and Logs.

Reviewer 3: @achatterjee-grafana, what do you think of putting this paragraph in a note?

<div class="clearfix"></div>

## Query the Azure Resource Graph service

Azure Resource Graph (ARG) is a service in Azure that is designed to extend Azure Resource Management by providing efficient and performant resource exploration with the ability to query at scale across a given set of subscriptions so that you can effectively govern your environment. By querying ARG, you can query resources with complex filtering, iteratively explore resources based on governance requirements, and assess the impact of applying policies in a vast cloud environment.

{{< figure src="/static/img/docs/azure-monitor/azure-resource-graph.png" class="docs-image--no-shadow" caption="Azure Resource Graph editor" max-width="650px" >}}

### Table queries

> Queries are written in the [Kusto Query Language](https://docs.microsoft.com/en-us/azure/governance/resource-graph/concepts/query-language). Please note that not all Kusto language features are available in ARG. An Azure Resource Graph query is formatted as table data.

Reviewer 2: ```suggestion
Queries are written in the [Kusto Query Language](https://docs.microsoft.com/en-us/azure/governance/resource-graph/concepts/query-language). Not all Kusto language features are available in ARG. An Azure Resource Graph query is formatted as table data.
```

<div class="clearfix"></div>

### Logs alerting

> Only available in Grafana v7.0+.

Grafana alerting is supported for Application Insights. This is not Azure Alerts support. Read more about how alerting in Grafana works in [Alerting rules]({{< relref "../alerting/_index.md" >}}).

> ## Query Application Insights service

Author: Application Insights and Insights analytics have been deprecated and cannot be used in new panels

    portalLink=properties.links.azurePortal
| summarize numberOfResources=count(resourceId) by tostring(recommendationName), tostring(recommendationState)
```

## Configure the data source with provisioning

> You can configure data sources using config files with Grafana’s provisioning system. You can read more about how it works and all the settings you can set for data sources on the [provisioning docs page]({{< relref "../administration/provisioning/#datasources" >}})

Reviewer 2: ```suggestion
You can configure data sources using config files with Grafana’s provisioning system. For more information on how it works and all the settings you can set for data sources on the [provisioning docs page]({{< relref "../administration/provisioning/#datasources" >}})
```

datasources:
  - name: Azure Monitor
    type: grafana-azure-monitor-datasource
    access: proxy
    jsonData:
      azureAuthType: msi
      subscriptionId: <subscription-id> # Optional, default subscription
    version: 1
```

### App Registration (client secret)

```yaml
datasources:
  - name: Azure Monitor
    type: grafana-azure-monitor-datasource
    access: proxy
    jsonData:
      azureAuthType: clientsecret
      cloudName: azuremonitor # See table below
      tenantId: <tenant-id>
      clientId: <client-id>
      subscriptionId: <subscription-id> # Optional, default subscription
    secureJsonData:
      clientSecret: <client-secret>
    version: 1
```

### Supported cloud names

| Azure Cloud                                      | Value                      |
| ------------------------------------------------ | -------------------------- |
| Microsoft Azure public cloud                     | `azuremonitor` (_default_) |
| Microsoft Chinese national cloud                 | `chinaazuremonitor`        |
| US Government cloud                              | `govazuremonitor`          |
| Microsoft German national cloud ("Black Forest") | `germanyazuremonitor`      |

> ## Deprecating Application Insights and Insights Analytics

Reviewer 2: ```suggestion
## Deprecated Application Insights and Insights Analytics
```

Application Insights and Insights Analytics are two ways to query the same Azure Application Insights data, which can also be queried from Metrics and Logs. In Grafana 8.0, Application Insights and Insights Analytics are deprecated and made read-only in favor of querying this data through Metrics and Logs. Existing queries will continue to work, but you cannot edit them. New panels are not able to use Application Insights or Insights Analytics.

For Application Insights, new queries can be made with the Metrics query type by selecting the "Application Insights" resource type.

{{< figure src="/static/img/docs/azure-monitor/app-insights-metrics.png" max-width="650px" class="docs-image--no-shadow" caption="Azure Monitor Application Insights example" >}}

For Insights Analaytics, new queries can be written with Kusto in the Logs query type by selecting your Application Insights resource.

{{< figure src="/static/img/docs/azure-monitor/app-insights-logs.png" max-width="650px" class="docs-image--no-shadow" caption="Azure Logs Application Insights example" >}}

> {{< figure src="/static/img/docs/azure-monitor/app-insights-resource-picker.png" max-width="650px" class="docs-image--no-shadow" caption="Azure Logs Application Insights resource picker" >}}

Reviewer 2: Here you have 2 images back to back, is this by design? Needs a sentence in between explaining the second image.
